cpu/samd5x/periph/can.c

```diff
@@ -824,223 +828,279 @@ static int _set(candev_t *candev, canopt_t opt, void *value, size_t value_len)
     return res;
 }
 
-static void _isr(candev_t *candev)
+static void _mcan_hdr_can_frame(struct can_frame * frame, uint32_t fe_r0, uint32_t fe_r1)
 {
-    can_t *dev = container_of(candev, can_t, candev);
+    /* while fifo element R0 and R1 in rx and tx are not the same
+     * the most relevant bits of r0 and r1 are shared between the different
+     * fifo element types this using the TXEFE definition to parse them */
+    canid_t canid = 0;
+    if (fe_r0 & CAN_TXEFE_0_XTD) {
+        canid |= CAN_EFF_FLAG;
+        canid |= (fe_r0 & CAN_TXEFE_0_ID_Msk) >> CAN_TXEFE_0_ID_Pos;
+    }
+    else {
+        /* 11 bit can id are stored in bit [28:18] of the ID field */
+        canid |= (fe_r0 & CAN_TXEFE_0_ID_Msk) >> CAN_TXEFE_0_ID_Pos >> 18;
+    }
 
-    uint32_t irq_reg = dev->conf->can->IR.reg;
-    DEBUG("isr: IR reg = 0x%08" PRIx32 "\n", irq_reg);
+    if (fe_r0 & CAN_TXEFE_0_RTR) {
+        canid |= CAN_RTR_FLAG;
+    }
+    frame->can_id = canid;
+    frame->can_dlc = (fe_r1 & CAN_TXEFE_1_DLC_Msk) >> CAN_TXEFE_1_DLC_Pos;
+
+    /* timestamp and message marker are currently unprased */
+}
 
```

Comment:
```suggestion
```

(One empty line [ought to be enough for anyone](https://quoteinvestigator.com/2011/09/08/640k-enough/).)